Improve displayed version name when building from PR (#8237)
readthedocs/readthedocs.org: docs/builds.rst

   :widths: 15, 10, 30

   ``READTHEDOCS``, Whether the build is running inside RTD, ``True``

>    ``READTHEDOCS_VERSION``, The RTD slug of the version which is being built, ``latest``

Reviewer: > I am not exactly sure if I understand it. What would it be the build context? Also not sure what it would mean to have "data without it being a nicer UX"... Sorry for that (it seems that you guys are used to some terms when talking about the project, but for me as an external contributor it is a bit difficult to understand).

This is the build context I'm talking about. We have a [proposed plan](https://docs.readthedocs.io/en/latest/development/design/theme-context.html) to change this, but we're now not 100% sure we want additional info here. What's the goal of providing this data?

Reviewer 2: These are environment variable, not the html sphinx context. This is so users can read these vars and do things like `html_title = f"{project} {READTHEDOCS_VERSION}"` https://github.com/readthedocs/readthedocs.org/pull/8237#issuecomment-856092333 (having some extra logic if the version is external). Having this as env var doesn't break the current html context, and it works for all tools. We can omit this change from this PR if you consider, and leave only the change from the version name on the footer.

Another addition for env vars would be https://github.com/readthedocs/readthedocs.org/issues/8260

Author: Thank you very much @stsewd for explaining the need for the env vars.

@ericholscher indeed my main reason for exposing these variables is what @stsewd pointed out. I hope to use the information provided by RTD to compose suitable titles/automatic texts in the sphinx templates. Expanding a bit on the example, these env vars would allow something like the following:
```python
if READTHEDOCS_VERSION_TYPE:
	html_title = f"{project} PR #{READTHEDOCS_VERSION_NAME}"
else:
	html_title = f"{project} {READTHEDOCS_VERSION_NAME}"
```
(the difference between the slug and the verbose name being pointed out in https://github.com/readthedocs/readthedocs.org/pull/8237#discussion_r660676316)

They should not interfere at all with the sphinx build process, but rather help the documentation authors to properly display certain properties in the documentation.

Reviewer 3: We are fine adding the environment variable here as this does not affect the Sphinx's build context and it's done in a generic and useful way.

   ``READTHEDOCS_VERSION_NAME``, Corresponding version name as displayed in RTD's version switch menu, ``stable``
   ``READTHEDOCS_VERSION_TYPE``, Type of the event triggering the build, ``branch`` ``tag`` ``external`` ``unknown``
   ``READTHEDOCS_PROJECT``, The RTD slug of the project which is being built, ``my-example-project``
   ``READTHEDOCS_LANGUAGE``, The RTD language slug of the project which is being built, ``en``

.. note::

   - The term slug is used to refer to a unique string across projects/versions containing ASCII characters only.
     This value is used in the URLs of your documentation.

   - If ``READTHEDOCS_VERSION_TYPE`` is ``external``,

>      it means that the version was built from pull request. See :doc:`/pull-requests`.

Reviewer 3: I'd say to move this text to the description of the variable itself instead of in a note. After that, we remove the bullets and keep just the initial paragraph of the note which is great.

   :widths: 15, 10, 30

   ``READTHEDOCS``, Whether the build is running inside RTD, ``True``
   ``READTHEDOCS_VERSION``, The RTD slug of the version which is being built, ``latest``
   ``READTHEDOCS_VERSION_NAME``, Corresponding version name as displayed in RTD's version switch menu, ``stable``
   ``READTHEDOCS_VERSION_TYPE``, Type of the event triggering the build, ``branch`` ``tag`` ``external`` ``unknown``
   ``READTHEDOCS_PROJECT``, The RTD slug of the project which is being built, ``my-example-project``
   ``READTHEDOCS_LANGUAGE``, The RTD language slug of the project which is being built, ``en``

.. note::

>    - The term "slug" is used here to indicate a string without some types characters
>      considered special in certain circunstances. The version slug can contain some
>      characters not usually present in other slugs.

Reviewer 2: Good to describe the slug :+1:, I don't think we explain that in any other place p:

```suggestion
   - The term slug is used refer to a unique string across projects/versions containing ASCII characters only.
     This value is used in the URLs of your documentation.
```

>    - The ``external`` value for ``READTHEDOCS_VERSION_TYPE`` will usually be
>      associated with builds triggered by pull requests. See :doc:`/pull-requests`.

Reviewer 2: ```suggestion
   - If ``READTHEDOCS_VERSION_TYPE`` is ``external``,
     it means that the version was built from pull request. See :doc:`/pull-requests`.
```
